### ISSUE TITLE
System.NullReferenceException in XmlCommentsModelFilter.ApplyPropertyComments

### ISSUE THREAD

**Issue by FreakyJ** [domaindrivendev/Ahoy#51]
Just yesterday I've tried this nice little tool and unfortunately I've got a NRE.

the controller code looks like this:

```javascript
public class Item
  {
    /// <summary>
    /// or that
    /// </summary>
    public int Id;
    /// <summary>
    /// or this
    /// </summary>
    public string Name;
  }

  /// <summary>
  /// Some summary
  /// </summary>
  /// 
  [Route("api/[Controller]")]
  [Produces("application/json", "text/json")]
  public class ItemsController : Controller
  {
    /// <summary>
    /// Return all items
    /// </summary>
    public static List<Item> Items = new List<Item>
    {
      new Item { Id = 1, Name = "First Test Item" },
      new Item { Id = 2, Name = "Second Test Item" },
    };

    /// <summary>
    /// Get all items
    /// </summary>
    /// <remarks>This will return all items</remarks>
    /// <returns>RETURN</returns>
    [HttpGet]
    public IEnumerable<Item> Get()
    {
      return Items;
    }

    /// <summary>
    /// Get Item by Id
    /// </summary>
    /// <param name="id">A var</param>
    /// <param name="dummy">just a dummy var</param>
    /// <returns>RETURN</returns>
    [HttpGet("itemId")]
    public Item Get([FromQuery]int id, int dummy)
    {
      return Items.FirstOrDefault(item => item.Id == id);
    }
  }
```
Using xml comments this will result in a Null reference Exception in XmlCommentsModelFilter.ApplyPropertyComments @ line 43
The problem is that memberInfo is null.

The problem seems to be with this part of the code above:
```javascript
public class Item
  {
    /// <summary>
    /// or that
    /// </summary>
    public int Id;
    /// <summary>
    /// or this
    /// </summary>
    public string Name;
  }
```
I fixed it by adding 
```javascript
if (memberInfo == null) return;
```

But I don't know if this is actually the proper way in doing it.

By the way: Will there be also a .Net4.5.1 version of Swashbuckle? :)
We plan to use it in MediaPortal2 and this is based on .Net4.5.1.

**Comment by domaindrivendev** [comment]
Fix should be available in the 6.0.0-beta9 release (targeting AspNet Core RC2). Let me know if this resolves your issue?

**Comment by hstk** [comment]
First, wanted to say that this is a great tool, thanks for putting it out there.

Running 6.0.0-beta9 on RC2, I'm finding some null reference exceptions around the same spot. This is the workaround I used to get running.

        public void Apply(Schema model, ModelFilterContext context)
        {
            var commentId = XmlCommentsIdHelper.GetCommentIdForType(context.SystemType);
            var typeNode = _xmlNavigator.SelectSingleNode(string.Format(MemberXPath, commentId));

            if (typeNode != null)
            {
                var summaryNode = typeNode.SelectSingleNode(SummaryTag);
                if (summaryNode != null)
                    model.Description = summaryNode.ExtractContent();
            }

            foreach (var entry in model.Properties)
            {
                var jsonProperty = context.JsonObjectContract.Properties[entry.Key];
                //if (jsonProperty == null) continue;
                if (jsonProperty.PropertyInfo() == null) continue;

                ApplyPropertyComments(entry.Value, jsonProperty.PropertyInfo());
            }
        }


![applypropertycomments](https://cloud.githubusercontent.com/assets/12897758/16018402/5d1dec26-3172-11e6-8521-128eac1bd796.jpg)
